gallery_dl/extractor/iwara.py

```diff
@@ -0,0 +1,357 @@
+            "username": user.get("username"),
+            "display_name": user.get("name").strip(),
+        }
+
+    def extract_media_info(self, item, key, include_file_info):
+        data = {
+            "id": item.get("id"),
+            "title": item.get("title").strip() if item.get("title") else "",
+        }
+
+        if include_file_info:
+            file_info = item if key is None else item.get(key, {})
+            filename = file_info.get("name")
+            filename, extension = filename.rsplit(".", 1)
+            createdAt = file_info.get("createdAt")
+            dt = text.parse_datetime(createdAt, "%Y-%m-%dT%H:%M:%S.%fZ")
+            datetime = dt.strftime(f"%a, %b {dt.day}, %Y %H:%M:%S")
+            data.update({
+                "file_id": file_info.get("id"),
+                "filename": filename,
+                "extension": extension,
+                "mime": file_info.get("mime"),
+                "size": file_info.get("size"),
+                "width": file_info.get("width"),
+                "height": file_info.get("height"),
+                "duration": file_info.get("duration"),
+                "datetime": datetime,
+                "type": file_info.get("type"),
+            })
+        return data
+
+    def get_metadata(self, user_info, media_info):
+        return {
+            **user_info,
+            **media_info
+        }
+
+    def yield_video(self, user_info, video):
+        video_info = self.extract_media_info(video, "file", True)
+        video_id = video_info.get("id")
+        file_id = video_info.get("file_id")
+        video = self.api.item(f"/video/{video_id}")
+        file_url = video.get("fileUrl")
+        sources = self.api.source(file_id, file_url)
+        source = next((r for r in sources if r.get("name") == "Source"), None)
+        download_url = source.get('src', {}).get('download')
+        url = f"https:{download_url}"
+        metadata = self.get_metadata(user_info, video_info)
+        yield Message.Directory, metadata
+        yield Message.Url, url, metadata
+
+    def yield_image(self, user_info, image_group):
+        image_group_info = self.extract_media_info(image_group, "file", False)
+        for image_file in image_group.get("files", {}):
+            image_file_info = self.extract_media_info(image_file, None, True)
+            image_info = {**image_file_info, **image_group_info}
+            file_id = image_info.get("file_id")
+            extension = image_info.get("extension")
+            url = (
+                f"https://i.iwara.tv/image/original/"
+                f"{file_id}/{file_id}.{extension}"
+            )
+            metadata = self.get_metadata(user_info, image_info)
+            yield Message.Directory, metadata
+            yield Message.Url, url, metadata
+
+
+class IwaraProfileExtractor(IwaraExtractor):
+    """Extractor for iwara.tv profile pages"""
+    subcategory = "profile"
+    pattern = BASE_PATTERN + r"/profile(?:/|$)"
```

Comment:
This would match URLs like `https://www.iwara.tv/profile`
```suggestion
    pattern = BASE_PATTERN + r"/profile/([^/?#]+)/?$"
```